scripts/test_reports/report.py

```diff
@@ -343,44 +369,405 @@ def discover(self):
+            if case_id:
+                tids.append(case_id)
+
+                result['case_id'] = case_id
+                del result['ref']
+                tmp_res.append(result)
+            else:
+                print("Cant find case for {}".format(ref))
+
+        entry = {
+            "include_all": False,
+            "case_ids": tids,
+            "config_ids": [result_file['id']],
+        }
+        self.final_results.append(
+            {'config': config, 'results': tmp_res, 'run_entry': entry, 'config_id': result_file['id']})
+
+    def discover(self):
+
+        results = {}
+        self.result_files.append({"file": self.results_file, "config": self.config})
+
+        # Get platforms from project
+        p = Configurations(project_id=self.project_id)
+        p.get("Platforms")
+
+        # See if we need to create new configurations on project
+        for rf in self.result_files:
+            config = rf['config']
+            if p.provides(config=config) == 0:
+                print("Need to create new config %s" % config)
+                p.add(config)
+
+        # update
+        p.get("Platforms")
+        for rf in self.result_files:
+            rf['id'] = p.provides(rf.get('config'))
+
 
 def parse_args():
     parser = argparse.ArgumentParser(
-                description="Upload results from TCF")
-    parser.add_argument('-j', '--junit-dir', default=None,
-            help="Directory with junit files")
+                description="Upload test results testrail")
+
+
+    parser.add_argument("--runner", default='sanitycheck', choices=['tcf', 'sanitycheck', 'maxwell', 'autopts'],
+                        help="""
+Select runner to import from.
+""")
 
-    parser.add_argument('-V', '--commit', default=None, help="Version being tested (git desribe string)")
+    result_file_select = parser.add_argument_group("Result files (input)",
+                                        """
+
+You can either select a directory with multiple files or just point to
+one file depending on the source of the results.
+
+TCF results expect a directory with multiple files,
+where sanitycheck expect 1 file per configuration.
+                                        """)
+
+
+    xor_results = result_file_select.add_mutually_exclusive_group()
+    xor_results.add_argument('-j', '--results-dir', default=None,
+            help="Directory with test result files")
+
+    xor_results.add_argument('-f', '--results-file', default=None,
+            help="File with test results format.")
+
+    parser.add_argument('-c', '--config', default=None,
+            help="Configuration name.")
+
+    parser.add_argument('-V', '--commit', default=None,
+        help="Version being tested (git desribe string)")
 
     parser.add_argument('-p', '--project', type=int, help="Project ID")
     parser.add_argument('-s', '--suite', type=int, help="Suite ID")
+
     parser.add_argument('-n', '--dry-run', action="store_true", help="Dry run")
```

Comment:
It seem --dry-run argument is not used. It still uploads test results with this --dry-run specified.